ZBaselines/Common/Scenario/BinaryClassification/BreastCancer/SaveModelToOnnxTest.json

```diff
@@ -0,0 +1,720 @@
+{
```

Comment:
Thanks @mzs for this PR, however, I think we should update the onnxml.cs as well. I don't think the ONNX model generated with this will pass the ONNX checker. at minimum we should add opset_import.

Reply:
There is a bug in ONNX repo that prevents proto3 files to be compiled for C#. Once that is resolved I will update it.

---
In reply to: [191226755](https://github.com/dotnet/machinelearning/pull/248#discussion_r191226755) [](ancestors = 191226755)

Reply:
I see: [https://github.com/onnx/onnx/issues/1045](https://github.com/onnx/onnx/issues/1045)

If they do not fix in timely fashion, we probably ought to use whatever the last version of the proto file can (1) produce valid models and (2) doesn't have this problem.

---
In reply to: [191536841](https://github.com/dotnet/machinelearning/pull/248#discussion_r191536841) [](ancestors = 191536841,191226755)

Reply:
Agreed. The current version that is checked-in is from 02/2018 and the last good version is from 03/2018. 

---
In reply to: [191592025](https://github.com/dotnet/machinelearning/pull/248#discussion_r191592025) [](ancestors = 191592025,191536841,191226755)

Reply:
onnx file to C#:

using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Windows.Media;
using Windows.Storage;
using Windows.AI.MachineLearning.Preview;

// SaveModelToOnnxTest

namespace App2
{
    public sealed class SaveModelToOnnxTestModelInput
    {
        public IList<float> Features { get; set; }
    }

    public sealed class SaveModelToOnnxTestModelOutput
    {
        public IList<float> Score { get; set; }
        public IList<float> Probability { get; set; }
        public IList<float> PredictedLabel { get; set; }
        public SaveModelToOnnxTestModelOutput()
        {
            this.Score = new List<float>();
            this.Probability = new List<float>();
            this.PredictedLabel = new List<float>();
        }
    }

    public sealed class SaveModelToOnnxTestModel
    {
        private LearningModelPreview learningModel;
        public static async Task<SaveModelToOnnxTestModel> CreateSaveModelToOnnxTestModel(StorageFile file)
        {
            LearningModelPreview learningModel = await LearningModelPreview.LoadModelFromStorageFileAsync(file);
            SaveModelToOnnxTestModel model = new SaveModelToOnnxTestModel();
            model.learningModel = learningModel;
            return model;
        }
        public async Task<SaveModelToOnnxTestModelOutput> EvaluateAsync(SaveModelToOnnxTestModelInput input) {
            SaveModelToOnnxTestModelOutput output = new SaveModelToOnnxTestModelOutput();
            LearningModelBindingPreview binding = new LearningModelBindingPreview(learningModel);
            binding.Bind("Features", input.Features);
            binding.Bind("Score", output.Score);
            binding.Bind("Probability", output.Probability);
            binding.Bind("PredictedLabel", output.PredictedLabel);
            LearningModelEvaluationResultPreview evalResult = await learningModel.EvaluateAsync(binding, string.Empty);
            return output;
        }
    }
}


---
In reply to: [191661419](https://github.com/dotnet/machinelearning/pull/248#discussion_r191661419) [](ancestors = 191661419,191592025,191536841,191226755)